doc/uftrace-live.md

```diff
@@ -992,6 +1000,121 @@ Each field in 'script_context' can be read inside the script.
 Please see `uftrace-script`(1) for details about scripting.
 
 
+AGENT
+=====
+uftrace supports running an agent inside the traced target, which can modify the
+tracing config at runtime.  The agent is disabled by default, and is enabled at
+start-up using the `-g`/`--agent` option.  The user can interact with the agent
+during while the target executes, from uftrace client instance, using the
+`-p`/`--pid` option.
+
+The client currently supports the following features:
+  * toggle tracing
+  * call depth filter
+  * time threshold filter
+  * opt-in and opt-out filters
+  * caller filters
+
+Consider the following program, which calls `a() -> b() -> c()` twice, and waits
+for external input in between.
+
+    $ cat abc_abc.c
+    void c(void) {
+        /* do nothing */
+    }
+
+    void b(void) {
+        c();
+    }
+
+    void a(void) {
+        b();
+    }
+
+    int main(void) {
+        a();
+        wait_for_sigusr1();
+        a();
+
+        return 0;
+    }
+
+    $ gcc -pg -o abc_abc abc_abc.c
+
+Tracing can be toggled anytime during execution.
+
+    $ uftrace --agent --trace=off abc_abc &
+    $ uftrace --pid $(pidof abc_abc) --trace=on
+    $ kill -s SIGUSR1 $(pidof abc_abc)
+    # DURATION     TID     FUNCTION
+      10.508 us [ 30324] |   } /* wait_for_sigusr1 */
+                [ 30324] |   a() {
+                [ 30324] |     b() {
+       0.138 us [ 30324] |       c();
+       0.757 us [ 30324] |     } /* b */
+       1.217 us [ 30324] |   } /* a */
+      12.346 us [ 30324] | } /* main */
+
+The call depth filter can be increased or decreased from the client.
+
+    $ uftrace --agent --depth=2 abc_abc &
+    $ uftrace --pid $(pidof abc_abc) --depth=4
+    $ kill -s SIGUSR1 $(pidof abc_abc)
+    # DURATION     TID     FUNCTION
+                [ 32384] | main() {
+       0.324 us [ 32384] |   a();
+       5.081  s [ 32384] |   wait_for_sigusr1();
+                [ 32384] |   a() {
+                [ 32384] |     b() {
+       0.106 us [ 32384] |       c();
+       0.552 us [ 32384] |     } /* b */
+       0.862 us [ 32384] |   } /* a */
+       5.081  s [ 32384] | } /* main */
+
+The time threshold can also be increased or decreased from the client.
+
+    $ uftrace --agent --time-filter=0.5us abc_abc &
+    $ uftrace --pid $(pidof abc_abc) --time-filter=0.1us
```

Comment:
The example output doesn't match to with filters exactly.  For more intuitive result, I think we can change it to start with 0.8us and then change it to 0.5us.